### ISSUE TITLE
"Directory" class provide confusing "." and ".." output 

### ISSUE THREAD

**Issue by Biliogadafr** [godotengine/godot#4761]
Godot 2.0.3 windows7x64

We have Directory class. (http://docs.godotengine.org/en/latest/classes/class_directory.html)

and along with normal output it produce '.' and '..' as available directories.

Code from documentation illustrates this behaviour.
```python
func dir_contents(path): 
	var dir = Directory.new()
	if dir.open(path) == OK:
		dir.list_dir_begin()
		var file_name = dir.get_next()
		while (file_name != ""):
			if dir.current_is_dir():
				print("Found directory: " + file_name)
			else:
				print("Found file: " + file_name)
			file_name = dir.get_next()
	else:
		print("An error occurred when trying to access the path.")
```
call  of  dir_contents("res://")

will make this result:

Found directory: .
Found directory: ..
Found directory: .git
Found file: arena.scn
Found directory: assets
...and so on

This may be unexpected when someone is trying to recursively walk over the directory tree (and ends with path like res://././././././.   )

Of course it is easy to workaround by adding some checks like 
if(file_name != ".." && file_name != "."):

I assume that something in godot relies on such behavior. But since 'Directory' class is part of public API for work with filesystem, this behaviour is quite confusing and output should not contain '.' and '..' by default.

**Comment by 29jm** [comment]
It's a pretty standard behavior I think, readdir in C does the same for example

**Comment by Biliogadafr** [comment]
Indeed... Ok. Then I personally see such behaviour for the first time :). In C++ I was using Boost (or dirent.h). Java filesystem api also doesn't behave like that.

**Comment by 29jm** [comment]
Godot seems to use dirent.h internally, so that's why :) It's mentionned in the docs, so IMO this is fine

**Comment by Biliogadafr** [comment]
oh... I had a look under the hood. Each platform is using it's own native api (and dirent.h for unix particularly:) ). Now I see that this is common behaviour for native APIs. But I still not sure that this is good behaviour for high level dynamic language. I didn't insist on this change anymore, but for me it is still archaic or too low-level. So it would be better if someone else will decide if this change is good/bad and if it worth the effort.

**Comment by Calinou** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by AlexHolly** [comment]
Maybe we can add something like this to the example in the docs?

```
if dir.current_is_dir():
                print("Found directory: " + file_name)
                if (file_name!=".") && (file_name!=".."):
                                # valid folder
```
